core/base/src/TString.cxx

```diff
@@ -1225,12 +1230,16 @@ void TString::Clear()
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Clear string and make sure it has a capacity of nc.
+/// \warning If nc > MaxSize(), then Fatal() is raised and only MaxSize() elements are allocated
+/// \return Resulting allocated capacity (after clamping, if needed)
 
-void TString::Clobber(Ssiz_t nc)
+Ssiz_t TString::Clobber(Ssiz_t nc)
 {
    if (nc > MaxSize()) {
-      Error("TString::Clobber", "capacity too large (%d, max = %d)", nc, MaxSize());
-      nc = MaxSize();
+      Fatal("TString::Clobber", "capacity too large (%d, max = %d)", nc, MaxSize());
+      UnLink();
+      Zero();
+      nc = MaxSize(); // Clamping after deleting to avoid corruption
```

Comment:
`Fatal` aborts the program in most cases, will the following be executed then?

Reply:
Ah I see, so you wrote the following lines for those edge cases where the users explicitly force `Fatal` to not be actually fatal right?

If so, maybe a comment about this would help, since this really is not the default behaviour and would happen extremely rarely